### PR TITLE
AAP-27923: implement EDA HA Cluster Redis support.

### PR DESCRIPTION
**Pull request by jshimkus-rh**
https://issues.redhat.com/browse/AAP-27923

### CODE REVIEW

File: src/aap_eda/settings/default.py [hunk 1]
```diff
@@ -403,14 +414,24 @@ def _rq_redis_client_additional_parameters():
...
     params = {k.lower(): v for (k, v) in params.items()}
     return params
 
 
+def rq_redis_client_instantiation_parameters():
+    params = rq_standalone_redis_client_instantiation_parameters()
+
+    # Include the HA cluster parameters.
+    params["clustered"] = bool(REDIS_HA_CLUSTER_HOSTS)
+    params["clustered_hosts"] = REDIS_HA_CLUSTER_HOSTS
```
Comment (dsavineau) [inline src/aap_eda/settings/default.py:430]: Not sure if you need to update the DAB version included in EDA but the clustered/clustered_hosts options were changed not so long ago

```suggestion
    if bool(REDIS_HA_CLUSTER_HOSTS):
        params["mode"] = "cluster"
    params["redis_hosts"] = REDIS_HA_CLUSTER_HOSTS
```

https://github.com/ansible/django-ansible-base/commit/6ff14f82046cfa565f9fdcc05f9ec4c1a2011bb6

Comment (jshimkus-rh) [reply src/aap_eda/settings/default.py]: @dsavineau Updated to latest.

File: src/aap_eda/settings/default.py [hunk 2]
```diff
@@ -403,14 +414,25 @@ def _rq_redis_client_additional_parameters():
...
     params = {k.lower(): v for (k, v) in params.items()}
     return params
 
 
+def rq_redis_client_instantiation_parameters():
+    params = rq_standalone_redis_client_instantiation_parameters()
+
+    # Include the HA cluster parameters.
+    if REDIS_HA_CLUSTER_HOSTS:
+        params["mode"] = "cluster"
```
Comment (mkanoor) [inline src/aap_eda/settings/default.py:430]: @jshimkus-rh Do we need a log line to indicate that we are in Cluster mode might help in debugging if we only have the logs available.

File: src/aap_eda/settings/default.py [hunk 3]
```diff
@@ -370,9 +372,18 @@ def _get_databases_settings() -> dict:
...
+REDIS_DB = settings.get("MQ_DB", DEFAULT_REDIS_DB)
 RQ_REDIS_PREFIX = settings.get("RQ_REDIS_PREFIX", "eda-rq")
 
+# The HA cluster hosts is a string of <host>:<port>[,<host>:port>]+
+# and is exhaustive; i.e., not in addition to REDIS_HOST:REDIS_PORT.
+# EDA does not validate the content, but relies on DAB to do so.
+#
+# In establishing an HA Cluster Redis client connection DAB ignores
+# the host and port kwargs.
+REDIS_HA_CLUSTER_HOSTS = settings.get("MQ_REDIS_HA_CLUSTER_HOSTS", "").strip()
```
[the remark this answers is not in the document]
Comment (jshimkus-rh) [reply src/aap_eda/settings/default.py:385]: @mkanoor 
A string as a sequence of `<host>:<port>[,<host>:<port>]*`.  
This is the form DAB takes them in, so other than stripping leading/trailing whitespace we pass it untransformed.

File: src/aap_eda/settings/default.py (hunk 2 above)
Comment (jshimkus-rh) [reply src/aap_eda/settings/default.py:430]: @mkanoor I was originally going to add one, but we aren't currently logging anything as part of default.py so I didn't.  If folks want it and are okay with it I can add it.

**Review by mkanoor — APPROVED** [review]

**Review by zkayyali812 — APPROVED** [review]

**Pull request merged**